ipn/ipnlocal/local.go

```diff
@@ -4729,7 +4730,34 @@ type tailFSTransport struct {
 	b *LocalBackend
 }
 
-func (t *tailFSTransport) RoundTrip(req *http.Request) (*http.Response, error) {
+func (t *tailFSTransport) RoundTrip(req *http.Request) (resp *http.Response, err error) {
+	bw := &requestBodyWrapper{}
+	if req.Body != nil {
+		bw.ReadCloser = req.Body
+		req.Body = bw
+	}
+
+	defer func() {
+		if req.Method != httpm.PUT && req.Method != httpm.GET {
+			return
+		}
+
+		i := strings.LastIndex(req.URL.String(), ".")
+		fileType := "unknown"
+		if i >= 0 && i < len(req.URL.String())-1 {
+			fileType = req.URL.String()[i+1:]
+		}
+		t.b.mu.Lock()
+		selfNodeKey := t.b.netMap.SelfNode.Key().ShortString()
+		t.b.mu.Unlock()
+		n, _, ok := t.b.WhoIs(netip.MustParseAddrPort(req.URL.Host))
+		shareNodeKey := "unknown"
+		if ok {
+			shareNodeKey = string(n.Key().ShortString())
+		}
+		t.b.logf(fmt.Sprintf("tailfs: access: %s from %s to %s: status code=%v type=%s tx=%d rx=%d", req.Method, selfNodeKey, shareNodeKey, resp.StatusCode, fileType, bw.bytesRead, resp.ContentLength))
```

Comment:
This `bytesRead` is useful on PUT. On GET, we want the bytesRead from the response. You could do this by:

1. Wrapping the response body
2. Writing the log only once the response body is closed, or Read from the response body returns an error

Reply:
Put differently, I think we want to report both the Content Length header (as a separate field), and the actual bytes downloaded as `rx`.